File: pcb.py
current_pid = 0


class PCB(object):

    def __init__(self, memory_start_region="", readwrite="",
                 file_name="", file_size=""):
        global current_pid
        self.pid = current_pid + 1
        current_pid += 1
        self.memory_start_region = memory_start_region
        self.readwrite = readwrite
        self.file_name = file_name
        self.file_size = file_size

    @classmethod
    def getCurrentPid(self):
        return current_pid

File: project2/PCB.scala

import scala.collection.mutable.ArrayBuffer

class PCB {
  PID.pid += 1
  var pid = PID.pid

  var memoryStartRegion: Int = 0

  var readwrite: String = ""

  var fileName: String = ""

  var fileSize: Float = 0.0f

  var cylinder: Int = 0

  var tau: Float = 0.0f

  var cpuTime: Float = 0.0f

  var tauLeft: Float = 0.0f

  var bursts: Float = 0.0f

  var burstCount: Int  = 0
}

class PCBRQOrdering extends Ordering[PCB] {
  override def compare(x: PCB, y: PCB): Int = x.tauLeft compare y.tauLeft
}

class PCBDiskOrdering extends Ordering[PCB] {
  override def compare(x: PCB, y: PCB): Int = x.cylinder compare y.cylinder
}

/**
 *
 * Singleton to hold a PID...
 */
object PID {
  var pid = -1
}